src/main/java/org/datadog/jenkins/plugins/datadog/clients/BatchSender.java

```diff
@@ -22,23 +24,31 @@ public class CompressedBatchSender<T> implements JsonPayloadSender<T> {
     private final Map<String, String> headers;
     private final int batchLimitBytes;
     private final Function<T, JSONObject> payloadToJson;
+    private final boolean compress;
 
-    public CompressedBatchSender(HttpClient httpClient,
-                                 String url,
-                                 Map<String, String> headers,
-                                 int batchLimitBytes,
-                                 Function<T, JSONObject> payloadToJson) {
+    public BatchSender(HttpClient httpClient,
+                       String url,
+                       Map<String, String> headers,
+                       int batchLimitBytes,
+                       Function<T, JSONObject> payloadToJson,
+                       boolean compress) {
         this.httpClient = httpClient;
         this.url = url;
-        this.headers = headers;
+        this.headers = new HashMap<>(headers);
         this.batchLimitBytes = batchLimitBytes;
         this.payloadToJson = payloadToJson;
+        this.compress = compress;
+
+        if (compress) {
+            this.headers.put("Content-Encoding", "gzip");
+        }
```

Comment:
Let's add a comment explaining that Datadog Agent previous version X does not support compression in the proxy endpoint.

Reply:
Added a comment